app/src/main/java/de/westnordost/streetcomplete/data/osm/OsmElementQuestType.kt

```diff
@@ -8,6 +8,18 @@ import de.westnordost.streetcomplete.data.osm.download.MapDataWithGeometryHandle
 
 interface OsmElementQuestType<T> : QuestType<T> {
 
+    val nameGivingTags: List<String> get() = listOf("name", "brand")
```

Comment:
Generally it is problematic (and thus not good style) to have different functions on an interface influence each other. For example, if I override `getTitleReplacements`, `nameGivingTags` is suddenly defunct. Also, the interface should always be as small as possible. You introduce basically two levels here - `nameGivingTags` being the slightly easier-to-use and `getTitleReplacements` being the slightly more powerful interface. I think this is more confusing than helpful.

I'd very much prefer to have *only* `getTitleReplacements`. Code-wise, it is pretty straightforward, not much more complicated to write than the `nameGivingTags`:

```kotlin
override fun getTitleReplacements(tags: Map<String, String>, typeName: Lazy<String?>) =
    arrayOf(tags["name"] ?: tags["brand"])
```